FabLabKasse/faucardPayment/magpos/magpos.py

```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/env python
+"""
+pymagpos -- MagnaCarta POS protocol (minimal robust implementation)
+"""
+
+import serial
+import codes
+import logging
+import time
+
+class MagposError(Exception):
+    """ Base Exception Class for this Module """
+    pass
+
+
+class ResponseError(MagposError):
+    """
+    ResponseError occur when the response to a command does not match the command's OK signal.
+    """
+    def __init__(self, function, code):
+        self.code = code
+        self.function = function
+        self.response = [code]
+
+    def store_rawdata(self, response):
+        """ Stores the raw response for evaluation purpose """
+        self.raw = response
+
+    def read_rawdata(self):
+        """
+        Returns the raw response data
+        :return: Raw response data
+        :rtype: list of ints
+        """
+        return self.raw
+
+
+    def __str__(self):
+        return ("[{0}] Unintended response received:{1}".format(self.function,
+                codes.desc.setdefault(self.code, self.code)))
+
+
+class TransactionError(MagposError):
+    """
+    TransactionError occur when the amount that has been decreased does not match the given amount
+    """
+    def __init__(self, card, old, new, amount):
+        self.card = card
+        self.old = float(old)/100
+        self.new = float(new)/100
+        self.amount = float(amount)/100
+
+    def __str__(self):
+        return "Difference in balance does not match the amount that should have been decreased.\
+               \nCard:{0}\t Amount:{1:.2f}\nOld:{2:.2f}\tNew:{3:.2f}"\
+                .format(self.card, self.amount, self.old, self.new)
+
+
+class ConnectionTimeoutError(MagposError):
+    """
+    ConnectionTimeoutError occur, when the connection between the USB/RS232 reader and the MagnaBox is broken
+    and/or the MagnaBox does not send a response message.
+    """
+    def __init__(self):
+        pass
+
+    def __str__(self):
+        return ("Serial connection to MagnaBox timed out (did not send command?)")
+
+
+class MagPOS:
+    """
+    MagPos Class implements functions to access payment features of the MagnaCarta-Security and Payment-System
+    """
+    def __init__(self, device):
+        """
+        Initializes the serial port communication on the given device port
+        :param device: serial port name
+        :type device: str
+        """
+        pass
+
+
+    def start_connection(self, retries=5):
+        """
+        Initializes the connection
+        :return: True if connection successful, False otherwise
+        :rtype: bool
+        :param retries: Max. Attempts to accomplish connection, Default value is 5
+        :type retries: int
+        """
+        return True

```

Comment:
I think it should inform the user that the payment method is unavailable and not crash. You could always throw these exceptions and catch them or allow all the things to be done, other then finalize the payment. Since we don't want to accept non existing money.